pages/public_cloud/ai_machine_learning/training_tuto_09_train_rock-paper-scissors/guide.en-gb.md

```diff
@@ -0,0 +1,263 @@
+
+## Instructions
+
+### Create object storage for data
+
+To train the model you'll need data and a place where to save the trained model.
+You can reuse the previous object storage used in the Notebook tutorial [Train YOLOv8 to play to "rock paper scissors"](/pages/public_cloud/ai_machine_learning/notebook_tuto_15_rock-paper-scissors/) or follow the step [Create Object Storage containers](/pages/platform/ai/notebook_tuto_15_rock-paper-scissors/) of this same tutorial.
+
+### Train your model
+
+> [!primary]
+>
+> All source code is available on the GitHub repository [ai-training-examples]().
+>
+
+To train the model, we will use AI Training. This powerful tool will allow you to automate your pipelines and build fine-tuning phases easily.
+
+AI Training allows you to train models directly from your own Docker images.
+
+First, you need to create a Python script that is in charge of doing the training.
+You can copy and paste the following code in a file named `train-rock-paper-scissors`:
+
+```python
+import ultralytics
+from ultralytics import YOLO
+import shutil
+
+#######################################################################################################################
+## 🎯 The aim of this script is to do transfert learning on YOLOv8 model.                                            ##
+## 💿 The data for train the model are in /workspace/data/rock-paper-scissors/                                       ##
+## 🧠 The train model are stored in /workspace/model/rock-paper-scissors/                                            ##
+#######################################################################################################################
+
+# ✅ Check configuration
+ultralytics.checks()
+
+# 🧠 Load a pretrained YOLO model
+model = YOLO('yolov8n.pt')
+
+# 💪 Train the model with new data ➡️ one GPU / 10 itérations (epochs)
+model.train(data='/workspace/data/rock-paper-scissors/data.yaml', device=0, epochs=50, verbose=True)
+
+# 💾 Save the model
+exportedMetaData = model.export()
+print('Model save to : ' + exportedMetaData)
+
+# ➡️ Copy the model to the object storage
+shutil.copy(exportedMetaData, '/workspace/model/rock-paper-scissors/')
+```
+
+Then, create a `requirements.txt` file to declare the Python dependencies:
+
+```
+ultralytics
+opencv-python-headless
+```
+
+Then, create a Dockerfile compliant with AI Training.
+You can copy and paste the following code in a file named `Dockerfile`:
+
+```docker
+FROM --platform=linux/x86_64 python:3.8
+
+WORKDIR /workspace
+ADD requirements.txt /workspace
+
+RUN apt-get update && apt-get install libgl1 -y
+RUN pip install -r requirements.txt
+
+ADD train-rock-paper-scissors.py /workspace
+
+# Mandatory to run the jobs in rootless mode
+RUN chown -R 42420:42420 /workspace
+
+CMD [ "python3"  , "/workspace/train-rock-paper-scissors.py"]
+```
+
+Then, build the Docker image and push it in the registry:
```

Comment:
```suggestion
Then, build the Docker image and push it in the shared registry:
```

Reply:
Maybe add a warning and info about the shared registry:

> [!warning]
> **Warning**
> The shared registry should only be used for testing purposes. Please consider creating and attaching your own registry. More information about this can be found [here](/pages/public_cloud/ai_machine_learning/gi_07_manage_registry).

You can find the address of your shared registry by launching this command:

```console
ovhai registry list
```

Then, log in on your shared registry with your usual OpenStack credentials:

```console
docker login -u <user> -p <password> <shared-registry-address>
```

You will then be able to build and push your image on this shared registry.